File: src/main/java/com/example/chatService/entity/DMRoom.java
package com.example.chatService.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.OffsetDateTime;

@Entity
@Builder
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "dm_rooms")
public class DMRoom {

    @Id
    @Column(length = 36)
    private String roomId;

    @Column(nullable = false, length = 36)
    private String userAId;
    @Column(nullable = false, length = 36)
    private String userBId;

    @Builder.Default
    private OffsetDateTime lastMessageTime = OffsetDateTime.now();
}

File: src/main/java/com/example/chatService/entity/DMRoom.kt
package com.example.chatService.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.OffsetDateTime

@Entity
@Table(name = "dm_rooms")
class DMRoom (
        @Id
        @Column(length = 36)
        var roomId : String? = null,

        @Column(nullable = false, length = 36)
        var userAId : String? = null,
        @Column(nullable = false, length = 36)
        var userBId : String? = null,

        var lastMessageTime : OffsetDateTime = OffsetDateTime.now()
) {}